src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java

```diff
@@ -13,17 +14,36 @@
 public class CategoryServiceImpl implements ICategoryService {
 
     @Autowired
-   private CategoryMapper categoryMapper;
+    private CategoryMapper categoryMapper;
     @Autowired
-   private CategoryRepository categoryRepository;
+    private CategoryRepository categoryRepository;
 
     public CategoryDTO addCategory(CategoryDTO categoryDto) {
-        try{
+        try {
             Category CategoryEntity = categoryMapper.categoryDtoToCategoryEntity(categoryDto);
             Category savedEntity = categoryRepository.save(CategoryEntity);
-            return categoryMapper.categoryEntityToCategoryDto(savedEntity);}
-        catch (Exception e){
+            return categoryMapper.categoryEntityToCategoryDto(savedEntity);
+        } catch (Exception e) {
             throw new ConflictException("There is already a category with this name " + categoryDto.getName());
         }
     }
-}
\ No newline at end of file
+
+    @Override
+    public CategoryDTO modifyCategory(Long categoryId, CategoryDTO categoryDto) {
+
+        if (categoryRepository.existsById(categoryId)) {
+
+            Category category = categoryRepository.getById(categoryId);
+            category = categoryMapper.categoryDtoToCategoryEntity(categoryDto);
+            Category result = categoryRepository.save(category);
+            CategoryDTO newCategoryDto = categoryMapper.categoryEntityToCategoryDto(result);
+
+            return newCategoryDto;
+
+        } else {
+            throw new NotFoundException("InvalidDto");
```

Comment:
El mensaje que se muestra dice InvalidDto? Ponele algo mas descriptivo, por ejemplo Category not found, o Category wih id provided not found, no se algo que se te ocurra